lib/segment/src/index/hnsw_index/hnsw.rs

```diff
@@ -181,13 +189,22 @@ impl HNSWIndex {
                 )
             };
 
+            let do_convert = LinkCompressionExperimentalSetting::from_env().convert_existing;
+
+            let graph_links_path = if !is_links_compressed && do_convert {
+                GraphLayers::convert_to_compressed(&graph_path, &regular_links, &compressed_links)?;
```

Comment:
@generall
There were an error here (`graph_links_path` == `graph_links.bin` vs `graph_path` == `graph.bin`). It did not caught by unit tests, but I got it testing manually by running qdrant with `__QDRANT_COMPRESSED_LINKS=2`.

I think these indirections with filename passing are bulky and error-prone, so I made an alternative approach here: #5749.